### PR TITLE
Catch RequestsException Instead Of Connection Errors To Allow For Ret…

### PR DESCRIPTION
**Pull request by rjduffner**
…ries During Timeouts

### CODE REVIEW

**Comment by rjduffner** [comment]
@dkuang1980 @tknickman @moshez

File: tests/test_request.py [hunk 1]
```diff
@@ -218,7 +217,7 @@ def test_request_request__server_down(self):
 
         with mock.patch('requests.sessions.Session.request') as mock_request:
             def request(*args, **kwargs):
-                raise ConnectionError("Server down!")
+                raise RequestException("Server down!")
```
Comment (moshez) [inline tests/test_request.py]: Why this change? `ConnectionError` still inherits from `RequestException`, so should still be caught. Maybe I don't understand something?

Comment (rjduffner) [reply tests/test_request.py]: Fair point, I will add one for Timeout as well to make sure that is caught instead of changing this to be the parent.

**Review by kstensland — APPROVED** [review]
